src/Monolog/Formatter/MongoDBFormatter.php

```diff
@@ -100,6 +102,13 @@ protected function formatException(\Exception $exception, $nestingLevel)
 
     protected function formatDate(\DateTime $value, $nestingLevel)
     {
-        return new \MongoDate($value->getTimestamp());
+        $seconds = (int) $value->format('U');
+        $milliseconds = (int) $value->format('u') / 1000;
+
+        if ($seconds < 0) {
+            return new UTCDateTime($seconds * 1000 - $milliseconds);
+        } else {
+            return new UTCDateTime($seconds * 1000 + $milliseconds);
+        }
```

Comment:
It was obvious that this date formatting needed an update, but I didn't read into the rest of the class enough to see how this differs from NormalizerFormatter. All we're doing here is storing MongoDB's native date type instead of a string date produced by NormalizerFormatter.